tcms/rpc/api/testcase.py

```diff
@@ -279,8 +284,37 @@ def filter(query=None):  # pylint: disable=redefined-builtin
     if query is None:
         query = {}
 
-    return list(
-        TestCase.objects.filter(**query)
+    for key, val in query.items():
+        if not key.startswith(
+            ("setup_duration", "testing_duration", "expected_duration")
+        ):
+            continue
+
+        try:
+            duration = parse_duration(val)
+        except TypeError:
+            # val isn't a string or byte-like object
+            # item is probably something like 'setup_duration__isnull=True'
+            continue
+
+        if duration is None:
+            continue
+
+        query[key] = duration
```

Comment:
I added this `for` loop because the values of the duration keys in the dict `query` will always be strings as xmlrpc can't marshal them if they are `datetime.duration` objects. The loop parses these strings (if they are present) into `datetime.duration` objects -- so that `QuerySet.filter()` won't choke on them.

Reply:
Since we don't need to filter by duration fields on the API side, then I'll take this out.